app/Payment.php

```diff
@@ -4,20 +4,20 @@
 
 use Illuminate\Database\Eloquent\Model;
 use NumberFormatter;
-use App\Ministry;
 
 class Payment extends Model
 {
-    public function amount(){
+    public function amount()
+    {
         return number_format($this->amount, 2, '.', ',');
     }
 
     /**
-     * Return ministry name;
+     * Return organization details
      */
-    public function ministry(){
-        $ministryCode = substr($this->payment_code, 0, 4); //ministry code is first 4 digits in a payment code
-        $ministry = Ministry::where('code', 'LIKE', "%$ministryCode%")->get()[0]->only(['shortname', 'name']);
-        return $ministry; //return array keyed ['name' => '', 'shortname' => '']
+    public function ministry()
+    {
```

Comment:
No, this is wrong. Leave it as it is and read the comment on the code.

The current Ministry method returns the shortname and name of the ministry so you can always use that. If you need something else returned you can use it like so:

`$ministry = Ministry::where('code', 'LIKE', "%$ministryCode%")->get()[0]->only(['shortname', 'name', 'id', etc]);`

Reply:
can i create another method ?